### ISSUE TITLE
About routing algorithm

### ISSUE THREAD

**Issue by jeongHwarr** [lalonderodney/SegCaps#22]
I really appreciate your project. 
I have a question about the routing algorithm.

https://github.com/lalonderodney/SegCaps/blob/c6b3f9e3e86b6e80652a76408fd6840c1ebe999b/capsule_layers.py#L319-L334

I am hard to understand this part. 
As far as I understand, 
`logit` -> [batch_num, input_height, input_width, num_capsules*num_atoms, num_capsule] 
`votes_trans` -> [num_atoms, batch, input_capsule, input_height, input_width, num_capsule]
`route` is result of softmax about logit with axis=-1(num_capsule) 

Then, What does `preactivate_unrolled` mean?
What does it mean to multiply `route` and `vote_trans`?

The parts under it are also hard to understand.
Could you explain this part more detail?

Thank you

**Comment by cheneeheng** [comment]
@jeonghwaYoo 

1. route is the routing coefficient "r" from eq(1) in the paper while votes_trans is the "u_hat" or "prediction vector".

2. The rest is basically an identical implementation of algorithm 1 in the paper. Although i believe that "c" in line 4 of algorithm1 is the same as "r" in line 5.

Thanks.
